runtime/parachains/src/inclusion.rs

```diff
@@ -444,6 +446,15 @@ impl<T: Config> Module<T> {
 					Error::<T>::NotCollatorSigned,
 				);
 
+				let validation_code = <paras::Module<T>>::validation_code_at(para_id, now, None)
```

Comment:
It'll be too expensive to do this hash on every backed candidate.

The changes to have a validation-code-by-hash map should make it possible to change `CurrentCode`, `FutureCode`, and `PastCode` maps to store only the validation code hash. It does mean that any alteration or read of those maps will require an extra storage lookup, but those happen primarily offchain where it is cheap.

If `current_code` returned a hash, it would make this check very cheap.

Reply:
indeed, the refactor is done